### DISCUSSION TITLE
v3.5.4

### DISCUSSION THREAD

**Discussion by RussKie** [gitextensions/gitextensions#9597] [General] [answered]
Maintenance v3.5 release
![Downloads](https://img.shields.io/github/downloads/gitextensions/gitextensions/v3.5.4/total.svg)

https://github.com/gitextensions/gitextensions/blob/release/3.5/GitUI/Resources/ChangeLog.md#version-354-25-sep-2021

<hr /><em>This discussion was created from the release <a href='https://github.com/gitextensions/gitextensions/releases/tag/v3.5.4'>v3.5.4</a>.</em>

**Comment by cristianst85** [comment]
Hello @RussKie, 

It seems that this [ChangeLog](https://github.com/gitextensions/gitextensions/blob/master/GitUI/Resources/ChangeLog.md) was not yet updated, which is the one GitExtensions is pointing to.

![image](https://user-images.githubusercontent.com/12585988/136080206-ac6f1070-9ca5-4e2c-aa20-d2bf6ff447bb.png)

Thanks!

**Reply by RussKie** [reply]
Fixed, thank you

**Comment by ADTC** [comment]
Kindly post the SHA256 checksum of releases, to make sure we're getting genuine and complete files without corruption 🙂

**Comment by vbjay** [comment, marked as answer]
Check the digital signature of the download. Build from appveyor causes it
to be signed.

**Reply by RussKie** [reply]
msi: sha-256="01e4ed009f71a89eb7f707bda48242fed0458f19a2f2122a2d1f4246f32c403a"
zip: sha-256="b1e2fb39319e80ec5da2af66ca62ab1f7b0b63f2bdadc2c9388ac67e2ae6f74b"

**Comment by SamuelSandvik** [comment]
Does anyone else have an issue with plugin manager missing? My coworker tried to install newest version and it didn't show up via menu "Plugins -> Plugin Manager".

**Reply by ADTC** [reply]
I also don't see such an option in the Plugins menu. It's my first time to install and I didn't add any new plugins beyond what the setup offers. Is the Plugin Manager a plugin too?

**Reply by SamuelSandvik** [reply]
It has been in older versions (I am still using 2.28). It is possible to start manually tough. https://github.com/gitextensions/gitextensions/issues/8794#issuecomment-961698373

**Comment by RussKie** [comment]
Can you please clarify "missing"? Also "*install newest version*" - how and
why?

**Reply by ADTC** [reply]
Does GitExtensions have a plugin manager? I also don't see such an option in the Plugins menu. It's my first time to install and I didn't add any new plugins beyond what the setup offers. Is the Plugin Manager a plugin too?

**Comment by bewillcott** [comment]
I am trying to run Git Extensions with mono under Fedora 34 Linux, and I get these two errors:

[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'GitUI.FontUtil' threw an exception. ---> System.DllNotFoundException: uxtheme.dll assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) System.NativeMethods.OpenThemeData(intptr,string)
  at GitUI.FontUtil..cctor () [0x00000] in <6105e56ad87545c18d528a1ed1b02ca4>:0 
   --- End of inner exception stack trace ---
  at GitUI.CommandsDialogs.SettingsDialog.Pages.FormChooseTranslation..ctor () [0x0000c] in <6105e56ad87545c18d528a1ed1b02ca4>:0 
  at (wrapper remoting-invoke-with-check) GitUI.CommandsDialogs.SettingsDialog.Pages.FormChooseTranslation..ctor()
  at GitExtensions.Program.RunApplication () [0x00044] in <93eac88792724f77b24f2a71f6234b4a>:0 
  at GitExtensions.Program.Main () [0x0012c] in <93eac88792724f77b24f2a71f6234b4a>:0 

Unhandled Exception:
System.TypeInitializationException: The type initializer for 'GitExtUtils.GitUI.DpiUtil' threw an exception. ---> System.DllNotFoundException: user32.dll assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) GitExtUtils.GitUI.DpiUtil.GetDC(intptr)
  at GitExtUtils.GitUI.DpiUtil..cctor () [0x00000] in <b7e666cf4e2d46218e2f4f90128e0feb>:0 
   --- End of inner exception stack trace ---
  at BugReporter.BugReportForm..ctor () [0x00088] in <28fc0ee8cb734f8aad2206a3b72fdadc>:0 
  at (wrapper remoting-invoke-with-check) BugReporter.BugReportForm..ctor()
  at BugReporter.Program.Main () [0x000b0] in <28fc0ee8cb734f8aad2206a3b72fdadc>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'GitExtUtils.GitUI.DpiUtil' threw an exception. ---> System.DllNotFoundException: user32.dll assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) GitExtUtils.GitUI.DpiUtil.GetDC(intptr)
  at GitExtUtils.GitUI.DpiUtil..cctor () [0x00000] in <b7e666cf4e2d46218e2f4f90128e0feb>:0 
   --- End of inner exception stack trace ---
  at BugReporter.BugReportForm..ctor () [0x00088] in <28fc0ee8cb734f8aad2206a3b72fdadc>:0 
  at (wrapper remoting-invoke-with-check) BugReporter.BugReportForm..ctor()
  at BugReporter.Program.Main () [0x000b0] in <28fc0ee8cb734f8aad2206a3b72fdadc>:0 

Does anyone know how to fix this?
It seems that user32.dll is a Windows only dll.  Is there a fix?

**Reply by gerhardol** [reply]
You can only use 2.x in mono

**Reply by hulsiejames** [reply]
Hi @gerhardol is there an alternative compiler for GitExtensions 3.x? or are mono users stuck with 2.x

Cheers

**Reply by gerhardol** [reply]
Mono supports 2.x only, supposedly wine can be used for later versions.

**Comment by bplatypus** [comment]
First, thanks to all contributors for this awesome software!

Just wanted to know if you switched the colors for remote & local branches? (if yes, how can I come back to the old colors -> green for remote branches and red for local branches)

**Reply by ADTC** [reply]
IDK, red for remote, green for local sounds better IMHO 😉

**Reply by bplatypus** [reply]
maybe but after using this soft for the past 5 years with green for remote and red for local, not a big fan of this change :-D

**Comment by RussKie** [comment]
No, the colours are changed to align with the git's colour scheme.
I believe it is possible to create a local theme and flip the colours, see
https://github.com/gitextensions/gitextensions/tree/master/GitUI/Themes#how-to-create-a-custom-theme

**Reply by bplatypus** [reply]
thank you

**Comment by piratedrinkingcrew** [comment]
Hello, I'm getting 2 hits on VirusTotal (false positives), and can't deploy the app.
https://www.virustotal.com/gui/file/01e4ed009f71a89eb7f707bda48242fed0458f19a2f2122a2d1f4246f32c403a?nocache=1
ClamAV: Win.Malware.Doina-9956863-0
Google: Detected

They are still present in the latest alpha release of Git Extensions. I did a backward check for all releases in the past year and the result was the same.

**Reply by cristianst85** [reply]
Indeed, the ClamAV flags the msi installer as positive. Most likely a false positive. ClamAV has a report page [here](https://www.clamav.net/reports/fp) through which we can report false positives. Maybe it adds more weight to the report if someone from the GE team submits it.

I am not sure about the other one (i.e., Google).

P.S. I forgot to post this:
![image](https://user-images.githubusercontent.com/12585988/192109394-e447ce2a-ee33-4fd7-84fb-6eec5208e105.png)

**Reply by vbjay** [reply]
A good test to see if it is appveyor.. build installer locally and run on that built installer. If still detected then most likely false positive or problem with build tools.

**Reply by vbjay** [reply]
Small hunch. Might have something to do with plugin manager. Might be seeing download dynamic code/programs to run and seeing it as virus like behavior.

**Reply by gerhardol** [reply]
The details reports that ClamAV finds ConEmu to be a Trojan.
Seem to be the same with the latest ConEmu alpha too: https://www.fosshub.com/ConEmu.html
Also the BugReporter is flagged for some reason.
There are some "suspicious" actions like accessing GitHub.com too...
PowerShell to handle Telemetry in the installer is flagged too.

Google also reports an issue. I do not get a warning when uploading the MSI, but when uploading a dev build to Google Drive, no reason why.

For the latest 4.0 alpha BugReporter is not flagged (only ConEmu).

For VirusTotal user voting, maintainers do not have special treatment, anyone can help.
Opening a discussion issue on ConEmu.
https://github.com/Maximus5/ConEmu/issues/2476

**Reply by cristianst85** [reply]
@gerhardol, I was suggesting submitting a false-positive report **directly to ClamAV**. I doubt that upvoting on VirusTotal will make the issue go away. It's good that we narrow it down to ConEmu which I can also confirm (it took me a little bit longer to build and scan).

![image](https://user-images.githubusercontent.com/12585988/192113943-83fee101-e38c-4e27-9944-ce5c5fdd7e29.png)

**Reply by gerhardol** [reply]
> @gerhardol, I was suggesting submitting a false-positive report **directly to ClamAV**. 

I had done that too.
Any recommended motivation?

> I doubt that upvoting on VirusTotal will make the issue go away

Meant that as an additional step.

**Reply by gerhardol** [reply]
ClamAV has now dropped the claim that ConEmu and by extension GE contains malware.
Google still raise the flag though. I have uploaded the files to my Drive, got a notification that the files violated Google policies, requested a review. No response, but it seems like I can share the link again today.

@christianst85
I meant if you or anyone else were aware of any good statement so the ClamAV processing would be simple. Maybe needed for Google.

**Reply by piratedrinkingcrew** [reply]
I upvoted on virustotal too, and re-scanned the app, the ClamAV is now gone as you said, and the app is passing the deployment virus scan. Thank you for you implication!

**Reply by cristianst85** [reply]
>I meant if you or anyone else were aware of any good statement so the ClamAV processing would be simple. Maybe needed for Google.

I don't think it matters. Most likely the false positive was due to a new _signature_ being added to the database.

Btw, the latest scan as of now (2022-09-27 17:15:04 UTC) shows that all red flags are gone (including the one from Google).

Proof.
![image](https://user-images.githubusercontent.com/12585988/192599124-1a58242a-7ba4-4064-b8eb-c6f5ad793203.png)
![image](https://user-images.githubusercontent.com/12585988/192599354-d8057ed3-6cf1-4d93-bc4c-f9634b01fd80.png)

**Comment by borysl** [comment]
Well ... I cannot open dialog with the commit in this version :(

**Reply by vbjay** [reply]
Providing details and screenshots would get better results than "It doesn't work" like comments.

**Comment by Juscc53** [comment]
- - - - [@]()_********_https://user-images.githubusercontent.com/115897010/196009311-257c162c-a144-4871-90d0-be85cf95cc7a.MOV
